packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -125,16 +125,12 @@ export class MetadataGenerator {
     }
 
     function _examinePaths(
-      tester: { paths: Array<{ isParam: boolean; path: string }>; method: Tsoa.Method },
-      subject: { paths: Array<{ isParam: boolean; path: string }>; method: Tsoa.Method },
+      tester: { paths: Array<{ cntParams: number; path: string }>; method: Tsoa.Method },
+      subject: { paths: Array<{ cntParams: number; path: string }>; method: Tsoa.Method },
     ): PathDuplicationType {
       const testLength = tester.paths.length > subject.paths.length ? subject.paths.length : tester.paths.length;
       for (let i = 0; i < testLength; i += 1) {
-        if (
-          (tester.paths[i].isParam && !subject.paths[i].isParam) ||
-          (!tester.paths[i].isParam && subject.paths[i].isParam) ||
-          (!tester.paths[i].isParam && tester.paths[i].path !== subject.paths[i].path)
-        ) {
+        if (tester.paths[i].cntParams !== subject.paths[i].cntParams || tester.paths[i].path !== subject.paths[i].path) {
```

Comment:
just came into my mind: I think we could probably even more simplify this check, cntParms is not exactly needed anymore, as I compare the path with empty placeholders now.

Reply:
simplified it